pkg/controller/controlplane/utils.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controlplane
+
+import (
+	"encoding/json"
+	"fmt"
+
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// DNSNamesForService returns the possible DNS names for a service with the given name and namespace
+func DNSNamesForService(name, namespace string) []string {
```

Comment:
All the functions here can be tested nicely, I guess that's what @adracus meant. :) Can we have a few tests for these utilities, please?

Reply:
Done.